### PR TITLE
Update search in navigation to expanding search

### LINKED ISSUE #4226: Old search still being used in Nav on vanilla docs
- Search should be updated to use the new expanding search
- Search should be in nav throughout all pages for consistency (currently it's only on docs and examples)

### PR DESCRIPTION
**Pull request by bethcollins92**
## Done

- Update `run` script
- Keep consistency with search box visible in header throughout all pages
- Update search box to use new expanding search

Fixes #4226 

## QA

- Open [demo](https://vanilla-framework-4228.demos.haus)
- Visit all pages in header
- Expand and close search, check it works as expected.
- Search for something and check it works and gives expected results

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical-web-and-design/vanilla-framework/blob/main/templates/docs/whats-new.md).
- [x] Documentation side navigation should be updated with the relevant labels.

### CODE REVIEW

File: templates/_layouts/_header.html [hunk 1]
```diff
@@ -19,13 +19,61 @@
...
+          <button type="submit" class="p-search-box__button" name="submit"><i class="p-icon--search">Submit</i></button>
+        </form>
+      </div>
     </nav>
   </div>
+  <div class="p-navigation__search-overlay"></div>
 </header>
+
+<script>
+  /** @param {HTMLElement} element The tab that acts as the handles. */
```
Comment (bartaz) [inline templates/_layouts/_header.html:42]: I guess we don't need that comment here, but also it doesn't seem relevant to search.

Was it copied from our search example? If so it seems we need to update it there too :)

File: templates/_layouts/_header.html [hunk 2]
```diff
@@ -19,13 +19,61 @@
...
+          <button type="reset" id="search-docs-reset" class="p-search-box__reset u-no-margin--right" name="close"><i class="p-icon--close">Clear input</i></button>
+          <button type="submit" class="p-search-box__button" name="submit"><i class="p-icon--search">Submit</i></button>
+        </form>
+      </div>
     </nav>
   </div>
+  <div class="p-navigation__search-overlay"></div>
 </header>
+
+<script>
```
Comment (bartaz) [inline templates/_layouts/_header.html:41]: As we will practically need this snippet everywhere on the site now, we should move it to JS file. I think we have some scripts.js in static files or somewhere.

File: templates/_layouts/_header.html (hunk 1 above)
Comment (bethcollins92) [reply templates/_layouts/_header.html]: Removed it from there!

File: templates/_layouts/_header.html (hunk 2 above)
Comment (bethcollins92) [reply templates/_layouts/_header.html]: I've added a conditional statement to fix the console error and added it to the script file

File: templates/_layouts/_header.html [hunk 3]
```diff
@@ -19,13 +19,23 @@
...
+          <button type="reset" id="search-docs-reset" class="p-search-box__reset u-no-margin--right" name="close"><i class="p-icon--close">Clear input</i></button>
+          <button type="submit" class="p-search-box__button" name="submit"><i class="p-icon--search">Submit</i></button>
+        </form>
+      </div>
     </nav>
   </div>
+  <div class="p-navigation__search-overlay"></div>
 </header>
+
+<script src="{{ versioned_static('js/scripts.js') }}"></script>
```
Comment (bartaz) [inline templates/_layouts/_header.html:41]: This will cause this file to be included twice in docs.

In `docs.html` template we already include it.
Now, that we need it on ALL pages, I think we should include it in the `_root.html` template (and remove from docs template).

**Review by bartaz — APPROVED** [review]
LGTM, thanks!
